app/services/github_handler.rb

```diff
@@ -35,50 +39,59 @@ def handle
     BAD_REQUEST
   end
 
-  def handle_review_request
-    owner = create_or_find_user(payload['pull_request']['user'])
-    pr = create_or_find_pr(payload['pull_request'])
-    # Even if you select multiple reviewers at once,
-    # the webhook sends a post for every person selected
-    # we can assume it is going to be a single user obj "requested_reviewer"
-    reviewer = create_or_find_user(payload['requested_reviewer'])
-    pr.review_requests.create!(data: payload, owner: owner, reviewer: reviewer)
+  def webhook_verified?
+    digest = OpenSSL::HMAC.hexdigest('SHA1', ENV['GITHUB_ANALYZER_WEBHOOK_SECRET'], raw_payload)
+    ActiveSupport::SecurityUtils.secure_compare("sha1=#{digest}", signature)
   end
 
-  def handle_review_removal
-    reviewer = User.find_by(github_id: payload['requested_reviewer']['id'])
-    pr = PullRequest.find_by(github_id: payload['pull_request']['id'])
-    pr.review_requests.find_by!(reviewer: reviewer)&.removed!
+  def opened
+    pr_data = payload.pull_request
+    PullRequest.create!(
+      node_id: pr_data.node_id,
+      number: pr_data.number,
+      state: pr_data.state,
+      locked: pr_data.locked,
+      draft: pr_data.draft,
+      title: pr_data.title,
+      body: pr_data.body,
+      closed_at: pr_data.closed_at,
+      merged_at: pr_data.merged_at,
+      merged: pr_data.merged,
+      github_id: pr_data.id
+    )
+  rescue ActiveRecord::RecordNotUnique
+    Rails.logger.error('This pull request already exists!')
```

Comment:
As I mentioned in the previous comment I think it's more correct to just let the exception happen if we are not going to handle it. This log will even give us less information on what happened than the exception

Reply:
Ok!